app/scripts/metamask-controller.js

```diff
@@ -2863,11 +2863,20 @@ export default class MetamaskController extends EventEmitter {
     this.controllerMessenger.subscribe(
       'NetworkController:networkDidChange',
       async () => {
-        this.txController.stopIncomingTransactionPolling();
+        const filteredChainIds = this.#getAllAddedNetworks().filter(
+          (networkId) =>
+            this.preferencesController.state.incomingTransactionsPreferences[
+              networkId
+            ],
+        );
+
+        if (filteredChainIds.length > 0) {
```

Comment:
I added back this check to ensure that we do not poll transactions if user has not set it for any network. This is fix while I am working on this PR: https://github.com/MetaMask/metamask-extension/pull/31269